From 30be68d825e462e93797ff6d0c55dcd5ad1d7a72 Mon Sep 17 00:00:00 2001
From: Simon Ser <contact@emersion.fr>
Date: Mon, 21 Nov 2022 18:42:47 +0100
Subject: [PATCH 1/2] Fix Wayland build
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes the following build error:

    vulkancapsviewer.cpp: In member function ‘bool VulkanCapsViewer::initVulkan()’:
    vulkancapsviewer.cpp:765:41: error: ‘wl_display_connect’ was not declared in this scope
      765 |             surfaceCreateInfo.display = wl_display_connect(NULL);
          |                                         ^~~~~~~~~~~~~~~~~~
---
 vulkancapsviewer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vulkancapsviewer.cpp b/vulkancapsviewer.cpp
index 711b4d3..d8721a9 100644
--- a/vulkancapsviewer.cpp
+++ b/vulkancapsviewer.cpp
@@ -60,6 +60,10 @@
 #include <QX11Info>
 #endif
 
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+#include <wayland-client.h>
+#endif
+
 #ifdef VK_USE_PLATFORM_ANDROID_KHR
 #include <QtAndroid>
 #include <QAndroidJniEnvironment>

From 83c234df361f474b704ed740c410f035b33f63a6 Mon Sep 17 00:00:00 2001
From: Simon Ser <contact@emersion.fr>
Date: Mon, 21 Nov 2022 18:41:32 +0100
Subject: [PATCH 2/2] Fix Wayland build in CI

Wayland wasn't actually enabled: the .pro file checks for uppercase
"WAYLAND", not lowercase "wayland".
---
 .github/workflows/build.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index 56da754..de0f909 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -72,7 +72,7 @@ jobs:
           echo ${TARGET_PLATFORM}
           PATH="/opt/qt512/bin:$PATH"
           CXX="clang++"
-          qmake DEFINES+=wayland CONFIG+=release PREFIX=/usr
+          qmake DEFINES+=WAYLAND CONFIG+=release PREFIX=/usr
           make INSTALL_ROOT=appdir install ; find appdir/
           wget -c -nv "https://github.com/probonopd/linuxdeployqt/releases/download/continuous/linuxdeployqt-continuous-x86_64.AppImage"
           chmod a+x linuxdeployqt-continuous-x86_64.AppImage
